cmd/thor/utils.go

```diff
@@ -440,6 +450,8 @@ func newP2PComm(ctx *cli.Context, repo *chain.Repository, txPool *txpool.TxPool,
 		BootstrapNodes:  fallbackBootstrapNodes,
 		RemoteBootstrap: remoteBootstrapList,
 		NAT:             nat,
+		AllowedPeers:    allowedPeers,
+		NoDiscovery:     len(ctx.StringSlice(allowedPeersFlag.Name)) > 0,
```

Comment:
Am I missing something or this `NoDiscovery` here is the same passed to [new](https://github.com/vechain/thor/blob/a82d2dbfe53c3568fabc1313bb7841179d0db2ac/cmd/thor/utils.go#L485) but then is always hardcoded as true [here](https://github.com/vechain/thor/blob/a82d2dbfe53c3568fabc1313bb7841179d0db2ac/p2psrv/server.go#L70)?

Reply:
We only [store/use](https://github.com/vechain/thor/blob/a82d2dbfe53c3568fabc1313bb7841179d0db2ac/p2psrv/server.go#L64) the `opts` [in our business logic to discover new peers ](https://github.com/vechain/thor/blob/a82d2dbfe53c3568fabc1313bb7841179d0db2ac/p2psrv/server.go#L122)

But you're right it does look confusing, I'll add a comment that the p2p server default discovery mechanism is not used per default.